web_m2x_options/static/src/js/form.js

```diff
@@ -87,7 +87,7 @@ odoo.define('web_m2x_options.web_m2x_options', function (require) {
             var is_string = typeof option === 'string'
             var is_bool = typeof option === 'boolean'
             if (is_string) {
-                return option === 'true' || option === 'True'
+                return option === 'true' || option === 'True' || option === '1'
```

Comment:
please just pyeval the string, which is the reason it works in core

Reply:
...pyeval or json?

Reply:
[pyeval](https://github.com/OCA/OCB/blob/10.0/addons/web/static/src/js/views/form_common.js#L662) is used to convert the options dictionary into a Javascript object, after which a simple [falsy/truthy test](https://github.com/OCA/OCB/blob/10.0/addons/web/static/src/js/views/form_common.js#L235) is used to determine the value, which is why it works in core.

Since I am dealing with the `options` object that is already filled using `pyeval`, maybe the correct option is then just to replace the `is_option_set` check by a falsy/truthy test?

Reply:
(not all `is_option_set` checks, but eg. [this one](https://github.com/OCA/web/blob/10.0/web_m2x_options/static/src/js/form.js#L400))

Reply:
but I wonder how strings end up there in the first place? If somebody writes `<... options="{'test': True}" />`, this should come out as boolean, not string, right? This seems to me we do the parsing somewhere else wrongly. Do you agree? So we should pyeval when we read this?

Reply:
it's not in the options attribute anyways as you write above which makes part of my reaction bullshit, but i stick to that we should look at parsing and not add stuff that happens to end up there as a string

Reply:
As an alternative I could do this: https://github.com/sunflowerit/web/pull/3. Do you have a preference?